flocker/acceptance/endtoend/test_dataset.py

```diff
@@ -35,6 +40,112 @@ def test_dataset_creation(self, cluster):
         """
         return create_dataset(self, cluster)
 
+    def _get_package_source(self, default_version=None):
+        """
+        Get the package source for the flocker version under test from
+        environment variables.
+
+        :param unicode default_version: The version of flocker to use
+            if not specified in environment variables.
+
+        :return: A ``PackageSource`` that can be used to install the version of
+            flocker under test.
+        """
+        env_vars = ['FLOCKER_ACCEPTANCE_PACKAGE_BRANCH',
+                    'FLOCKER_ACCEPTANCE_PACKAGE_VERSION',
+                    'FLOCKER_ACCEPTANCE_PACKAGE_BUILD_SERVER']
+        missing_vars = list(x for x in env_vars if x not in os.environ)
```

Comment:
Yuck.. Surprising that list comprehensions are broken like that. FWIW, Python3 behaviour is different (correct)